icu4c/source/common/lstmbe.cpp

```diff
@@ -366,35 +378,36 @@ LSTMData::LSTMData(UResourceBundle* rb, UErrorCode &status)
     int32_t embedding_size = ures_getInt(embeddings_res.getAlias(), &status);
     LocalUResourceBundlePointer hunits_res(
         ures_getByKey(rb, "hunits", nullptr, &status));
+    if (U_FAILURE(status)) return;
     int32_t hunits = ures_getInt(hunits_res.getAlias(), &status);
     const UChar* type = ures_getStringByKey(rb, "type", nullptr, &status);
+    if (U_FAILURE(status)) return;
```

Comment:
yes

Reply:
yes, that is why

Reply:
I meant that you don't need the check on line 381, since the next two lines will be no-ops if the status is failed. (But it's not a big issue to check the status there anyways).

Reply:
If line 379 fail and I didn't check on 381, then hunits_res.getAlias() could be nullptr, right? I try not to call ures_getInt with the first parameter as nullptr.

Reply:
In `ures_getInt` it will first check if status is failed, or if the first argument is null. So if hunits_res.getAlias() is nullptr, then calling it should be fine. Though it might change the error status if it isn't already failed.
So maybe this is fine as-is.